### PR TITLE
[TASK] Refactor Link Rendering

### PR DESCRIPTION
**Pull request by linawolf**
Delegate link rendering to a dedicated LinkRenderer in preparation for integrating references into text roles.

### CODE REVIEW

File: lib/HTML/Renderers/LinkRenderer.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\RST\HTML\Renderers;
+
+use Doctrine\RST\Environment;
+
+final class LinkRenderer extends \Doctrine\RST\Renderers\LinkRenderer
```
Comment (greg0ire) [inline lib/HTML/Renderers/LinkRenderer.php]: I'm surprised phpcs is OK with this :thinking:

File: lib/HTML/Renderers/LinkRenderer.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public function __construct(Environment $environment)
+    {
+        parent::__construct($environment);
+    }
```
Comment (greg0ire) [inline lib/HTML/Renderers/LinkRenderer.php]: What am I missing here?

File: lib/HTML/Renderers/LinkRenderer.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    /** @param mixed[] $attributes */
+    public function renderUrl(?string $url, string $title, array $attributes = []): string
+    {
+        $url = (string) $url;
```
Comment (greg0ire) [inline lib/HTML/Renderers/LinkRenderer.php]: Let's inline this on line 22?

File: lib/LaTeX/Renderers/LinkRenderer.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+use function is_string;
+use function substr;
+
+final class LinkRenderer extends AbstractLinkRenderer
+{
+    public function __construct(Environment $environment)
+    {
+        parent::__construct($environment);
+    }
```
Comment (greg0ire) [inline lib/LaTeX/Renderers/LinkRenderer.php]: To be removed?

File: lib/HTML/Renderers/LinkRenderer.php (hunk 1 above)
Comment (linawolf) [reply lib/HTML/Renderers/LinkRenderer.php]: php cs was not ok with naming the latter LinkRendererInterface...

Comment (greg0ire) [reply lib/HTML/Renderers/LinkRenderer.php]: Oh right. How about aliasing it to `LinkRendererInterface` at import time?

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @linawolf !

**Pull request merged**